server/server.go

```diff
@@ -851,7 +861,114 @@ func (s *Server) handleExit(ctx context.Context, event fsnotify.Event) {
 	if err := s.ContainerStateToDisk(ctx, c); err != nil {
 		log.Warnf(ctx, "Unable to write %s %s state to disk: %v", resource, c.ID(), err)
 	}
+	if s.config.EventedPLEG {
+		err := s.generateCRIEvent(ctx, c, types.ContainerEventType_CONTAINER_STOPPED_EVENT)
+		if err != nil {
+			log.Warnf(ctx, "Unable to generate event %s for container %s due to err %s", types.ContainerEventType_CONTAINER_STOPPED_EVENT, c.ID(), err)
+		}
+	}
 	if err := os.Remove(event.Name); err != nil {
 		log.Warnf(ctx, "Failed to remove exit file: %v", err)
 	}
 }
+
+func (s *Server) getSandboxStatuses(ctx context.Context, sandboxId string) (*types.PodSandboxStatus, error) {
+	sandboxStatusRequest := &types.PodSandboxStatusRequest{PodSandboxId: sandboxId}
+	sandboxStatus, err := s.PodSandboxStatus(ctx, sandboxStatusRequest)
+
+	if IsNotFound(err) {
+		return nil, err
+	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	return sandboxStatus.GetStatus(), nil
+}
+
+func (s *Server) getContainerStatuses(ctx context.Context, sandboxUid string) ([]*types.ContainerStatus, error) {
+	listContainerRequest := &types.ListContainersRequest{Filter: &types.ContainerFilter{LabelSelector: map[string]string{"io.kubernetes.pod.uid": sandboxUid}}}
```

Comment:
Where does the label selector come from?

Reply:
We already vendor k8s pkg/kubelet/types and use it in a few places, including server/*.go. Why not to just `import kubetypes "k8s.io/kubernetes/pkg/kubelet/types"` in server.go and use the const name `kubetypes.KubernetesPodUIDLabel` instead of the duplicated constant value ?